session/src/main/java/org/apache/iotdb/session/Session.java

```diff
@@ -1978,14 +1986,170 @@ public void createSchemaTemplate(
       List<List<String>> measurements,
       List<List<TSDataType>> dataTypes,
       List<List<TSEncoding>> encodings,
-      List<CompressionType> compressors)
+      List<List<CompressionType>> compressors)
       throws IoTDBConnectionException, StatementExecutionException {
     TSCreateSchemaTemplateReq request =
         getTSCreateSchemaTemplateReq(
             name, schemaNames, measurements, dataTypes, encodings, compressors);
     defaultSessionConnection.createSchemaTemplate(request);
   }
 
+  public void createSchemaTemplate(
+      String name,
+      List<List<String>> measurements,
+      List<List<TSDataType>> dataTypes,
+      List<List<TSEncoding>> encodings,
+      List<List<CompressionType>> compressors)
+      throws IoTDBConnectionException, StatementExecutionException {
+    List<String> emptySchemaNames = new ArrayList<>();
+    TSCreateSchemaTemplateReq request =
+        getTSCreateSchemaTemplateReq(name, null, measurements, dataTypes, encodings, compressors);
+    defaultSessionConnection.createSchemaTemplate(request);
+  }
+
+  /**
+   * Construct Template at session and create it at server.
+   *
+   * @see Template
+   */
+  public void createSchemaTemplate(Template template)
+      throws IOException, IoTDBConnectionException, StatementExecutionException {
+    TSCreateSchemaTemplateReq req = getTSCreateSchemaTemplateReq();
+    req.setName(template.getName());
+    ByteArrayOutputStream baos = new ByteArrayOutputStream();
+    template.serialize(baos);
+    req.setSerializedTemplate(baos.toByteArray());
+    baos.close();
+    defaultSessionConnection.createSchemaTemplate(req);
+  }
+
+  public void addAlignedMeasurementsInTemplate(
```

Comment:
public interface should add java doc

Reply:
java docs about create, prune and query the template has been added now.